src/osd/OSD.cc

```diff
@@ -9400,6 +9400,26 @@ bool OSDService::_recover_now(uint64_t *available_pushes)
   return true;
 }
 
+unsigned OSDService::get_target_pg_log_entries()
+{
+  auto num_pgs = osd->get_num_pgs();
+  auto target = cct->_conf->osd_target_pg_log_entries_per_osd;
+  if (num_pgs > 0 && target > 0) {
```

Comment:
I am wondering if we should break these cases into two, where we could just fall back to osd_max_pg_log_entries when target is not set - this gives us a simple way to get back to the old behavior.

Reply:
that's what i was going to do, but then i realized that we probably want min to be lower than 3000 and max to be higher than 3000.  so... i could introduce a new osd_fallback_pg_log_entries maybe to use when the per-osd target is zeroed?

Reply:
min should still be lower with your change to the default value of osd_min_pg_log_entries, this will just ensure that we fall back to at least osd_max_pg_log_entries=3000 when we are in any of these https://github.com/ceph/ceph/pull/32683/files#diff-44b273f2b3311e9f70f3be99ef20070fL4047-L4052 and per-osd target is zeroed.

Reply:
sorry, i didn't parse that last comment.

with the current patch, min = 250 and max = 10000.  and target / 100 = 3000.  if we want it to tall back to 3000 when the per-osd target is 0, then i think we need another setting.. or we need to keep max at 3000.

note that the other change in behavior here is that before, min = max = 3000, so the PeeringState.cc code that uses min for clean PGs and max for dirty ones used to always be 3000, but is not either 250 or target/num_pgs (~3000).